### DISCUSSION TITLE
Multiple relationships

### DISCUSSION THREAD

**Discussion by perroudsky** [prisma/prisma#5903] [Q&A] [answered]
I'm having some difficulties setting up my schema for multiple relationships. Could not find any similar exemple in the docs. I'd like a tryptic like such. I get lost in the relation names, fields, etc... help would be greatly appreciated :)

```prisma
model Attribute {
  id          String  @id @default(cuid())
  name        String  @unique
description String?
}
```

```prisma
model User {
...
favoriteMajorAttribute Attribute[]
favoriteMinorAttribute Attribute[]

}
```

```prisma
model Post {
  ...
majorAttribute Attribute[]
minorAttribute Attribute[]
}
```

**Comment by ryands17** [comment, marked as answer]
@perroudsky 👋 

If you have multiple relationships, you need to specify a relation name to [disambiguate relations](https://www.prisma.io/docs/concepts/components/prisma-schema/relations#disambiguating-relations)

So the above schema would look like:

```prisma
model Attribute {
  id              String  @id @default(cuid())
  name            String  @unique
  description     String?
  userMajorAttr   User?   @relation("userMajorAttr", fields: [userMajorAttrId], references: [id])
  userMajorAttrId String?
  userMinorAttr   User?   @relation("userMinorAttr", fields: [userMinorAttrId], references: [id])
  userMinorAttrId String?
}

model User {
  id                     String      @id @default(cuid())
  favoriteMajorAttribute Attribute[] @relation("userMajorAttr")
  favoriteMinorAttribute Attribute[] @relation("userMinorAttr")
}
```

**Reply by perroudsky** [reply]
Hi @ryands17 ! thanks for your answer but I have not been clear enough sorry!


In my case, I would like users to have possible the same attributes.
Not one attribute exclusive to a user.
and the attribute needs to be a model because it has field like 'description', 'icon'...

Fox exemple:

User1 {
 favoriteMajorAttribute: [Yellow,**Blue**]
 favoriteMinorAttribute: [Red]
}
User2 {
 favoriteMajorAttribute: [Red, **Blue**]
 favoriteMinorAttribute: [Black]
}

Post {
majorAttribute: [Red, Blue],
minorAttribute: [Purple,Black,]
}

Is it more clear?

**Reply by perroudsky** [reply]
Ok this seems to work
```prisma

model Attribute {
  id              String  @id @default(cuid())
  name            String  @unique
  description     String?
  userMajorAttr   User[]   @relation("userMajorAttr")
  userMinorAttr   User[]   @relation("userMinorAttr")

}

model User {
  id                     String      @id @default(cuid())
  favoriteMajorAttribute Attribute[] @relation("userMajorAttr")
  favoriteMinorAttribute Attribute[] @relation("userMinorAttr")
}

**Reply by ryands17** [reply]
Ohh, so what you wanted was a many-to-many relation. I would suggest going through [this](https://www.prisma.io/docs/concepts/components/prisma-schema/relations) page in the docs for all types of relations explained and how to create them in Prisma :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
